e2e/single-cluster/imagescan_test.go

```diff
@@ -92,3 +68,211 @@ var _ = Describe("Image Scan", Label("infra-setup"), func() {
+
+			By("pushing a new tag to the registry and checking the new image tag is found in the bundle")
+			newTag := "0.0.0-50"
+			imageTag = tagAndPushImage("k8s.gcr.io/pause", image, newTag)
+			Eventually(func() string {
+				out, _ := k.Get("bundles", "imagescan-examples", "-o", "yaml")
+				return out
+			}).Should(
+				MatchRegexp(fmt.Sprintf(`image: %s # {"\$imagescan": "test-scan"}`, imageTag)))
+
+			By("checking that the new tag is pushed in the git repository")
+			err := gh.CheckoutRemote(repository, "imagescan")
+			Expect(err).NotTo(HaveOccurred())
+			basedir := filepath.Join(clonedir, "examples")
+			b, err := os.ReadFile(filepath.Join(basedir, "deployment.yaml"))
+			Expect(err).NotTo(HaveOccurred())
+			Expect(string(b)).Should(ContainSubstring(newTag))
+		})
+	})
+
+	// this test didn't pass before adding fix for issue #2096
+	When("deploy imagescan setup with pre-release images and * semver range", func() {
+		BeforeEach(func() {
+			assetdir = "imagescan/pre-releases-ignored"
+			tmpRepoDir = GinkgoT().TempDir()
+			image, imageTag = initRegistryWithImageAndTag("k8s.gcr.io/pause", "0.0.0-40")
+			applyTemplateValues(assetdir, tmpRepoDir, image, imageTag)
+		})
+		It("updates the image reference", func() {
+			By("checking the deployment exists")
+			Eventually(func() string {
+				out, _ := k.Namespace("default").Get("pods")
+				return out
+			}).Should(ContainSubstring("pause-prerelease-"))
+
+			By("checking the bundle has the original image tag")
+			Eventually(func() string {
+				out, _ := k.Get("bundles", "imagescan-examples", "-o", "yaml")
+				return out
+			}).Should(
+				MatchRegexp(fmt.Sprintf(`image: %s # {"\$imagescan": "test-scan"}`, imageTag)))
+
+			By("pushing a new tag to the registry and checking the fleet controller does not crash")
+			// store number of fleet controller restarts to compare later
+			fleetControllerRestarts := getFleetControllerRestarts(k)
+			newTag := "0.0.0-50"
+			previousImageTag := imageTag
+			imageTag = tagAndPushImage("k8s.gcr.io/pause", image, newTag)
+
+			// the scan time interval is 5 seconds.
+			// we check for 10 seconds so we're sure that the image has been scanned and the controller didn't crash
+			Consistently(func() int {
+				return getFleetControllerRestarts(k)
+			}, 10*time.Second, 1*time.Second).Should(
+				Equal(fleetControllerRestarts))
+
+			By("checking the bundle has the original image tag")
+			Eventually(func() string {
+				out, _ := k.Get("bundles", "imagescan-examples", "-o", "yaml")
+				return out
+			}).Should(
+				MatchRegexp(fmt.Sprintf(`image: %s # {"\$imagescan": "test-scan"}`, previousImageTag)))
+		})
+	})
+	AfterEach(func() {
+		_, _ = k.Delete("gitrepo", "imagescan")
+		_, _ = k.Delete("secret", "git-auth")
+	})
+})
+
+func getFleetControllerRestarts(k kubectl.Command) int {
+	out, err := k.Namespace("cattle-fleet-system").Get("pods", "-l", "app=fleet-controller", "--no-headers")
```

Comment:
If we didn't want to deal with indices, we could do something like this:
```suggestion
	out, err := k.Namespace("cattle-fleet-system").Get("pods", "-l", "app=fleet-controller", "--no-headers", "-o", "custom-columns=RESTARTS:.status.containerStatuses[0].restartCount")
```

Reply:
The `fleet-controller` has 3 containers so... are we sure 0 is going te be the container we're looking for? 
As I wasn't sure I used a global restart count.